xoa/data_source_hosts.go

```diff
@@ -0,0 +1,86 @@
+package xoa
+
+import (
+	"github.com/ddelnano/terraform-provider-xenorchestra/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+)
+
+func dataSourceXoaHosts() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHostsRead,
+		Schema: map[string]*schema.Schema{
+			"master": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"hosts": &schema.Schema{
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeMap,
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+			},
+			"pool": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
```

Comment:
This can reuse the existing `resourceTags` [schema](https://github.com/terra-farm/terraform-provider-xenorchestra/blob/6d0b85c152094efd9ee16ad0c8b6a080b701c305/xoa/tags.go#L5)